src/User32.Desktop/User32+WINDOW_STATION_ACCESS_MASK.cs

```diff
@@ -0,0 +1,308 @@
+        /// /// This is a set of Specific Rights bitmask that defines access rights and security for desktop objects
+        /// </summary>
+        /// <remarks>
+        /// <see cref="Kernel32.ACCESS_MASK"/> is quite well described here: http://blogs.msdn.com/b/openspecification/archive/2010/04/01/about-the-access-mask-structure.aspx
+        /// Window Station security and access rights are quite well described here: https://msdn.microsoft.com/en-us/library/windows/desktop/ms687391(v=vs.85).aspx
+        /// </remarks>
+        public struct WINDOW_STATION_ACCESS_MASK : IComparable, IFormattable, IComparable<WINDOW_STATION_ACCESS_MASK>, IEquatable<WINDOW_STATION_ACCESS_MASK>, IComparable<Kernel32.ACCESS_MASK>, IEquatable<Kernel32.ACCESS_MASK>
+        {
+            /// <summary>
+            /// Bits 28-31
+            /// </summary>
+            private const uint GenericRightsMask = 0xf0000000;
+
+            /// <summary>
+            /// Bits 24-27
+            /// </summary>
+            private const uint SpecialRightsMask = 0x0f000000;
+
+            /// <summary>
+            /// Bits 16-23
+            /// </summary>
+            private const uint StandardRightsMask = 0x00ff0000;
+
+            /// <summary>
+            /// Bits 0-15
+            /// </summary>
+            private const uint SpecificRightsMask = 0x0000ffff;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="WINDOW_STATION_ACCESS_MASK"/> struct.
+            /// </summary>
+            /// <param name="value">The value for the <see cref="WINDOW_STATION_ACCESS_MASK"/>.</param>
+            public WINDOW_STATION_ACCESS_MASK(uint value)
+            {
+                this.Value = value;
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="WINDOW_STATION_ACCESS_MASK"/> struct.
+            /// </summary>
+            /// <param name="value">The reference <see cref="Kernel32.ACCESS_MASK"/> for the <see cref="WINDOW_STATION_ACCESS_MASK"/>.</param>
+            public WINDOW_STATION_ACCESS_MASK(Kernel32.ACCESS_MASK value)
+            {
+                this.Value = value;
+            }
+
+            /// <summary>
+            /// Generic access rights for the interactive window station object, which is the window station assigned to the logon session of the interactive user.
+            /// </summary>
+            [Flags]
+            public enum InteractiveWindowStationGenericRight : uint
+            {
+                GENERIC_READ = Kernel32.ACCESS_MASK.StandardRight.STANDARD_RIGHTS_READ |
+                    SpecificRight.WINSTA_ENUMDESKTOPS |
+                    SpecificRight.WINSTA_ENUMERATE |
+                    SpecificRight.WINSTA_READATTRIBUTES |
+                    SpecificRight.WINSTA_READSCREEN,
+
+                GENERIC_WRITE = Kernel32.ACCESS_MASK.StandardRight.STANDARD_RIGHTS_WRITE |
+                    SpecificRight.WINSTA_ACCESSCLIPBOARD |
```

Comment:
To be true, I don't really know if these two documents refer to something that is declared on code (I don't think so), if it is something that is implicitly done by windows or if in order to have it you have to enable that Generic Right + those Specific Rights. I am not much knowledgeable of the Windows security mechanism and internals and to be true, it is the first time that I need to deal with it.

This is the way I understood should be done (explicitly setting the generic right + the specific rights for that object to semantically achieve the left column meaning for that object). If you have somewhere to point me towards I will happily read/research to understand it better. I didn't find much information on these kind of things on official documentation, just the references for the functions and types, some samples and this security page, but it assumes you are well versed on Windows Security as far as I understood it.

I might not be a Windows Internals hacker/nerd but I think I understand these native things and Windows API a little more than the average developer and the Desktop/Window Station information that I found (either official or on blogs/forums, etc) are really fuzzy on some areas. Although the API seem really simple I found out this is not a trivial thing to mess with on Windows, also it is relatively new in comparison to other API sets (this is since Windows Vista only) and it seems to be a moving target (we have the virtual desktops on Windows 8.x and the new desktop/metro apps thing that got in and out). I didn't find anything so far on the virtual desktops from Windows 8.x on official documents as  well as it has been very hard to find decent documentation on Windows 10 introduced APIs and changes.